gtsfm/loader/colmap_loader.py

```diff
@@ -0,0 +1,187 @@
+"""Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+Authors: John Lambert
+"""
+
+import os
+from pathlib import Path
+from typing import Optional
+
+from gtsam import Cal3Bundler, Pose3
+
+import gtsfm.utils.images as img_utils
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+logger = logger_utils.get_logger()
+
+
+class ColmapLoader(LoaderBase):
+    """Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+    Note: assumes all images are of the same dimensions.
+    TODO: support max-resolution constraint for images are of the same dimensions.
+       Would introduce a constraint between get_image() and get_intrinsics()
+
+    Folder layout structure:
+    - RGB Images: images/
+    - Intrinsics data (optional): cameras.txt (see https://colmap.github.io/format.html#cameras-txt)
+    - Extrinsics data (optional): images.txt (see https://colmap.github.io/format.html#images-txt)
+
+    If explicit intrinsics are not provided, the exif data will be used.
+    """
+
+    def __init__(
+        self,
+        colmap_files_dirpath: str,
+        images_dir: str,
+        use_gt_intrinsics: bool = True,
```

Comment:
If true, does it use colmap output as ground truth? In that case, I think it should be called use_colmap_result_as_gt.

Reply:
the data could be actual ground truth, just in the colmap format. So I think it's better this way. It's up to the user if we want to use COLMAP output as pseudo-ground truth.

For example, we save our GTSFM output in the COLMAP-format.